internal/sync/core.go

```diff
@@ -78,8 +79,24 @@ func (s *DefaultSynchronizer) Apply(ctx context.Context, reterr *error) {
 	}
 }
 
+func setFinalizers(obj client.Object) {
+	finalizers := obj.GetFinalizers()
+	if finalizers == nil {
+		finalizers = []string{}
+	}
+
+	// Only append the desired finalizer if it doesn't exist
+	for _, finalizer := range finalizers {
+		if finalizer == metav1.FinalizerDeleteDependents {
+			return
+		}
+	}
+
+	finalizers = append(finalizers, metav1.FinalizerDeleteDependents)
+	obj.SetFinalizers(finalizers)
+}
+
 func setOwnerReference(owner, obj client.Object) {
-	obj.SetFinalizers([]string{metav1.FinalizerDeleteDependents})
```

Comment:
This kept removing the `provider.cluster.x-k8s.io` finalized added by the `cluster-api-provider`

Reply:
Isn’t this change alone enough to fix the issue?

Reply:
Not this alone, there is another instance where we set the finalizers, that is in `patchStatus`.

Also we are experiencing multiple conflicts, this solves the issue with the *Providers resources managed by the cluster-api-operator, however the other commit that changes `client.Apply` to `client.Merge` is to prevent all resources from having a new `resourceVersion` on empty patches. This does apply to all syncers, providers, secrets, etc.

Also the reason I kept commits separate.

Reply:
It seems the tests are more stable with this change. I opened a separate [one](https://github.com/rancher/turtles/pull/916) to address infinite reconciliation issue with secrets, which is an issue with comparing the diff.

The only issue with this workflow is that we will cause resourceVersion bumps each time we patch CAPIProvider resource or operator resource. SSA was ment to avoid that, but it works [incorrectly](https://kubernetes.io/docs/concepts/configuration/secret/#basic-authentication-secret)  with `stringData` field. Therefore they have a diff comparison. Now the problem is on regular resources, where it is desirable to perform a diff check on the resource content before sending patch request.